src/libcore/convert.rs

```diff
@@ -99,28 +98,20 @@ use fmt;
 #[inline]
 pub const fn identity<T>(x: T) -> T { x }
 
-/// A cheap reference-to-reference conversion. Used to convert a value to a
-/// reference value within generic code.
-///
-/// `AsRef` is very similar to, but serves a slightly different purpose than,
-/// [`Borrow`].
+/// Used to do a cheap reference-to-reference conversion.
+/// This trait is similar to [`AsMut`] which is used for converting between mutable references.
+/// If you need to do a costly conversion it is better to implement [`From`] with type
+/// `&T` or write a custom function.
 ///
-/// `AsRef` is to be used when wishing to convert to a reference of another
-/// type.
-/// `Borrow` is more related to the notion of taking the reference. It is
-/// useful when wishing to abstract over the type of reference
-/// (`&T`, `&mut T`) or allow both the referenced and owned type to be treated
-/// in the same manner.
-///
-/// The key difference between the two traits is the intention:
 ///
+/// `AsRef` is very similar to, but serves a slightly different purpose than [`Borrow`]:
 /// - Use `AsRef` when the goal is to simply convert into a reference
```

Comment:
putting another `///` before this line to give the list some room to breathe would be great.